File: dotnet/src/Microsoft.Agents.Validation/Validators/ContentValidators/TextReasoningContentValidator.cs
using Microsoft.Agents.Abstractions.Models;

namespace Microsoft.Agents.Validation.Validators.ContentValidators;

/// <summary>
/// Validator for TextReasoningContent.
/// </summary>
public class TextReasoningContentValidator : ContentValidatorBase<TextReasoningContent>
{
    public override ValidationResult Validate(TextReasoningContent content, ValidationContext? context = null)
    {
        // TODO: Implement specific validation rules for TextReasoningContent
        return ValidationResult.Success();
    }
}

File: typescript/packages/agents-protocol-validation/src/validators/content/TextReasoningContentValidator.ts
import type { TextReasoningContent } from '@microsoft/agents-protocol-abstractions';
import type { ValidationContext } from '../../core/ValidationContext';
import { ContentValidatorBase } from '../base/ContentValidatorBase';
import { ValidationResult } from '../../core/ValidationResult';

/**
 * Validator for TextReasoningContent.
 */
export class TextReasoningContentValidator extends ContentValidatorBase<TextReasoningContent> {
  public validate(content: TextReasoningContent, context?: ValidationContext): ValidationResult {
    // TextReasoningContent has no strict validation requirements
    return ValidationResult.success();
  }
}
